pkg/controller/builder.go

```diff
@@ -185,3 +195,60 @@ func replaceImageHook(placeholder string, desiredImageEnvVar string) deploymentc
 		return []byte(newDeployment), nil
 	}
 }
+
+func updateEnv(con *corev1.Container, name, value string) ([]corev1.EnvVar, error) {
+	for _, e := range con.Env {
+		if e.Name == name {
+			return con.Env, fmt.Errorf("unexpected environment variable %q=%q in container %q while building manifests", e.Name, e.Value, con.Name)
+		}
+	}
+	if value == "" {
+		return con.Env, nil
+	}
+	klog.FromContext(context.Background()).WithName("builder").V(1).Info("Updated environment", "container", con.Name, "key", name, "value", value)
+	return append(con.Env, corev1.EnvVar{Name: name, Value: value}), nil
+}
+
+func setProxyEnvs(con *corev1.Container, vars map[string]string) error {
```

Comment:
Based on the function signature, I'd suggest a more generic name. Maybe `updateEnvFromMap`?